src/modalities/models/coca/coca_model.py

```diff
@@ -68,17 +83,26 @@ def forward(self, inputs: Dict[str, torch.Tensor]) -> Dict[str, torch.Tensor]:
 
 
 class MultiModalBlock(nn.Module):
-    def __init__(self, config: GPTConfig):
+    def __init__(
+        self,
+        n_embd: int,
+        bias: bool,
+        epsilon: float,
+        activation: ActivationType,
+        n_head: int,
+        dropout: float,
+        ffn_hidden: int,
+    ):
         super().__init__()
-        self.ln_1 = LayerNorm(ndim=config.n_embd, bias=config.bias, epsilon=config.epsilon)
-        self.attn = Attention(config.n_embd, config.n_head, config.bias, is_causal=True, use_cross_attention=True)
-        self.ln_2 = LayerNorm(ndim=config.n_embd, bias=config.bias, epsilon=config.epsilon)
-
-        if config.activation == ActivationType.GELU:
-            self.mlp = TransformerMLP(config)
-        elif config.activation == ActivationType.FUSED_SWIGLU:
-            hidden_dim = 256 * ((int(2 * 4 * config.n_embd / 3) + 256 - 1) // 256)
-            self.mlp = xops.SwiGLU(config.n_embd, hidden_dim, config.n_embd, bias=False)
+        self.ln_1 = LayerNorm(ndim=n_embd, bias=bias, epsilon=epsilon)
+        self.attn = Attention(n_embd, n_head, bias, is_causal=True, use_cross_attention=True)
+        self.ln_2 = LayerNorm(ndim=n_embd, bias=bias, epsilon=epsilon)
+
+        if activation == ActivationType.GELU:

```

Comment:
@mali-git for the sake of avoiding code duplication, would it helpful to somehow extract the selection of the activation or even the entire block as a whole? I am asking because the implementation is almost identical to the one in the [GPT2 Model](https://github.com/Modalities/modalities/blob/feat/coca/src/modalities/models/gpt2/gpt2_model.py#L196) except for the attention layer.